Fixes issue #177

NiteshKant opened the pull request:
- Provided a ChannelWriter.close(boolean flush) method that provides a close without flush functionality.
- Added a flush in onChannelReadComplete() for ServerRequestResponseConverter.

With this change, if the server's RequestHandler calls response.close(false) then the flush will only be done when the channel read completes.

Commit:
Fixes issue #177

- Provided a ChannelWriter.close(boolean flush) method that provides a close without flush functionality.
- Added a flush in onChannelReadComplete() for ServerRequestResponseConverter.

With this change, if the server's RequestHandler calls response.close(false) then the flush will only be done when the channel read completes.

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #108](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/108/) FAILURE
Looks like there's a problem with this pull request

Commit:
Fixed failing test.

Any server that processes the request in a different thread, should flush explicitly.

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #109](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/109/) FAILURE
Looks like there's a problem with this pull request

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #110](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/110/) SUCCESS
This pull request looks good
